x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java

```diff
@@ -230,15 +265,21 @@ public static String extractRequiredString(
         String scope,
         ValidationException validationException
     ) {
-        String requiredField = ServiceUtils.removeAsType(map, settingName, String.class);
+        int initialValidationErrorCount = validationException.validationErrors().size();
```

Comment:
This refactoring is here we have passed a  `ValidationException` into the function so we expect that to collect the errors but `removeAsType` would throw an ElasticsearchStatusException meaning there are 2 paths for validation errors to propagate.